### PR TITLE
chore: IRD-1696 conventional commits and release workflows

### PR DESCRIPTION
**Pull request by araad**
- Conventional commits in checks
- Create PR workflow
- Release workflow

Note: Doc will be added in another PR

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+16.20.2
```
Comment (zzal) [inline .nvmrc:1]: Got that from foundation, but is it ok if it's only v16 instead of v18?

**Review by zzal — APPROVED** [review]
Just a question

File: .nvmrc (hunk 1 above)
Comment (MPIngeno) [reply .nvmrc:1]: Good question

File: .github/workflows/create-release-pr.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        id: updated-version
+        shell: bash
+        run: |
+          version=$(node -p 'require("./lerna.json").version')
+          echo "Current project version is: $version"
+          echo "version=$version" >> $GITHUB_OUTPUT
+
+      - name: Checks if tag with the current project version already exists
+        uses: mukunku/tag-exists-action@v1.4.0
+        id: tag
```
Comment (twistdmnd) [inline .github/workflows/create-release-pr.yml:46]:
```suggestion
        id: checkForExistingTag
```

File: .github/workflows/pr-opened.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+jobs:
+  lint-commits:
+    name: lint-commits
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```
Comment (twistdmnd) [inline .github/workflows/pr-opened.yml:13]: I don't think we need fetch depth here 🤔 ❓

File: .github/workflows/pr-opened.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
```
Comment (twistdmnd) [inline .github/workflows/pr-opened.yml:18]: I'm not sure we need to enforce this, I think the yarn install below will simply use the nvmrc file. But I might be mistaken ❓

File: .github/workflows/release-actions.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        env:
+          NPM_READ_PACKAGES_TOKEN: ${{ secrets.NPM_READ_PACKAGES_TOKEN }}
+
+      - name: Get current project version
+        id: updated-version
+        shell: bash
+        run: |
+          version=$(node -p 'require("./lerna.json").version')
+          echo "Current project version is: $version"
+          echo "version=$version" >> $GITHUB_OUTPUT
```
Comment (twistdmnd) [inline .github/workflows/release-actions.yml:33]:
```suggestion
          echo "CURRENT_PROJECT_VERSION =$version" >> $GITHUB_OUTPUT
```

File: .github/workflows/release-actions.yml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+      - name: Configure Git User
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+
+      - name: Create GitHub Release
+        uses: ncipollo/release-action@v1.13.0
```
Comment (twistdmnd) [inline .github/workflows/release-actions.yml:41]: another one! nice 😄

File: actions/package.json [hunk 7]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "@ingeno-foundation/github-actions",
+  "version": "1.0.0",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/ingeno/foundation-github-actions.git",
+    "directory": "./actions"
+  },
+  "description": ""
```
Comment (twistdmnd) [inline actions/package.json:9]: I would add a descripton here or simply remove it from the package.json 📖

File: .github/workflows/pr-opened.yml (hunk 3 above)
Comment (zzal) [reply .github/workflows/pr-opened.yml:13]: It's copied from `foundation`… 📋

File: .github/workflows/release-actions.yml (hunk 6 above)
Comment (zzal) [reply .github/workflows/release-actions.yml:41]: No merit here, 'twas all copied form `foundation`

File: .github/workflows/pr-opened.yml (hunk 3 above)
Comment (twistdmnd) [reply .github/workflows/pr-opened.yml:13]: Okay, still I don't think it's required for this use case. Wanna try it ?

Comment (MPIngeno) [reply .github/workflows/pr-opened.yml:13]: Confirmed, we need it otherwise the workflow fails

File: .github/workflows/release-actions.yml [hunk 8]
```diff
@@ -0,0 +1,48 @@
+name: Release-actions
+on:
+  pull_request:
+    types:
+      - closed
+
+permissions:
+  contents: write
+  packages: write
```
Comment (jonathanpelletier) [inline .github/workflows/release-actions.yml:9]: Not sure we need the `packages` permission in this repo as we're just tagging

**Pull request merged**